libcloudforensics/providers/aws/internal/s3.py

```diff
@@ -118,3 +128,41 @@ def Put(self, s3_path: str, filepath: str) -> None:
           'Could not upload file {0:s}: {1:s}'.format(
               filepath, str(exception)),
           __name__) from exception
+
+  def GCSToS3(self,
+              project_id: str,
+              gcs_path: str,
+              s3_path: str) -> None:
+    """Copy an object in GCS to an S3 bucket.
+
+    (Creates a local copy of the file in a temporary directory)
+
+    Args:
+      project_id (str): Google Cloud project ID.
+      gcs_path (str): File path to the source GCS object.
+          Ex: gs://bucket/folder/obj
+      s3_path (str): Path to the target S3 bucket.
+          Ex: s3://test/bucket
+    Returns:
+      Dict: An API operation object for an S3 Put request.
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.put_object  # pylint: disable=line-too-long
+    Raises:
+      ResourceCreationError: If the object couldn't be uploaded.
+    """
+    gcs = gcp_storage.GoogleCloudStorage(project_id)
+    if not s3_path.startswith('s3://'):
+      s3_path = 's3://' + s3_path
+    if not gcs_path.startswith('gs://'):
+      gcs_path = 'gs://' + gcs_path
+    localcopy = gcs.GetObject(gcs_path)
```

Comment:
Is there any way that we can `stat` a remote GCS file? I'd like to output a message to the user warning them about the data transfer.

Reply:
PTAL

Reply:
How much checking to we wanna do here? Given that this might often be used to shuffle full disks around, would it be relevant to check if there's enough space at the destination directory to dump the file in?

Reply:
Great idea. It now checks generally during any "get" operation, throughout the download, if there's enough space.